packages/oauth/oauth_server.js

```diff
@@ -135,7 +135,41 @@ OAuth._checkRedirectUrlOrigin = redirectUrl => {
   );
 };
 
-const middleware = (req, res, next) => {
+// Parses request payload to use in the handler
+//
+// @param {Object} req incoming oauth request
+//
+const parseRequestData = req => new Promise((resolve, reject) => {
+  if (req.method === 'POST') {
+    let rawBody = '';
+    req
+      .on('data', (chunk) => {
+        rawBody += chunk;
+      })
+      .on('end', () => {
+        try {
+          const contentType = req.headers['content-type'];
+          let json = {};
+          if (contentType && contentType === 'application/x-www-form-urlencoded') {
+            json = querystring.parse(rawBody);
+          } else {
+            json = JSON.parse(rawBody);
+          }
+
+          resolve(json);
+        } catch (error) {
+          reject(error);
+        }
+      })
+      .on('error', e => reject(e.message));
+  } else {
+    resolve(req.query);
+  }
+});
+
+const middleware = async (req, res, next) => {
```

Comment:
You made this async, but there's nowhere an await.

Reply:
Made it clearer where we where waiting by using an await instead of then.